src/webgldrawer.js

```diff
@@ -1325,6 +1327,78 @@
             return shaderProgram;
         }
 
+        /**
+         * Test webgl rendering in the current browser by drawing a small image
+         * in a new viewer. If all pixel data is 0, rendering failed, and fallback
+         * to the CanvasDrawer is required for all tiled images.
+         * @private
+         * @returns
+         */
+        _doRenderingTest(){
+            // Avoid infinite call stack by only testing once
+            if( testing ){
+                return;
+            }
+            // Set the testing flag to true until the test viewer has been created
+            testing = true;
+
+            // A dataUrl with non-zero pixels to use
+            const dataUrl = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAAAXNSR0IArs4c6QAAAIRlWElmTU0AKgAAAAgABQESAAMAAAABAAEAAAEaAAUAAAABAAAASgEbAAUAAAABAAAAUgEoAAMAAAABAAIAAIdpAAQAAAABAAAAWgAAAAAAAABIAAAAAQAAAEgAAAABAAOgAQADAAAAAQABAACgAgAEAAAAAQAAAASgAwAEAAAAAQAAAAQAAAAANlaTAQAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDYuMC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KGV7hBwAAAElJREFUCB1jZACCa5uWJr/78JHVhkNkNuPpmhy/b5+Z1/74/vPzr2/3y1junTr5QYf1NAsDI4Pg1ivin0E6GLbnMDhsTmHwArEBM20b4j0ge4MAAAAASUVORK5CYII=";
+
+            // Create a small element for the test viewer and append it to our viewer's element
+            const div = document.createElement('div');
+            div.setAttribute('id', 'osd_internal_test');
+            div.style.width = '4px';
+            div.style.height = '4px';
+            div.style.position = 'fixed';
+            div.style.top = 0;
+            div.style.left = 0;
+            this.viewer.element.appendChild( div );
```

Comment:
Is this going to be visible to the user for a brief flash? Can we avoid that? Maybe give it opacity 0 or something?

Reply:
I haven't seen it flash, but I have now given it a very small opacity (0.0001). I didn't make it truly zero, just in case some browser optimizes away rendering if there is no opacity.

Reply:
Perfect :)